### PR TITLE
Add kotlin code snippets for Android DataStore

### PR DESCRIPTION
**Pull request by raphkim**
*Issue #, if available:*

*Description of changes:*
Add kotlin docs to datastore guide

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/lib/datastore/fragments/android/real-time/observe-snippet.md [hunk 1]
```diff
@@ -9,3 +12,21 @@ Amplify.DataStore.observe(Post.class,
...
+```kotlin
+Amplify.DataStore.observe(Post::class.java,
+    { Log.i("GetStarted", "Observation began.") },
+    { postChanged ->
+        val post = postChanged.item
+        Log.i("GetStarted", "Post: $post")
+    },
+    { failure -> Log.e("GetStarted", "Observation failed.", failure) },
+    { Log.i("GetStarted", "Observation complete.") }
+)
```
Comment (jamesonwilliams) [inline docs/lib/datastore/fragments/android/real-time/observe-snippet.md:28]: Does this work?
```
Amplify.DataStore.observe(Post::class.java,
    { Log.i("GetStarted", "Observation began.") },
    { Log.i("GetStarted", "Post: $(it.item())") },
    { Log.e("GetStarted", "Observation failed.", it) },
    { Log.i("GetStarted", "Observation complete.") }
)
```

File: docs/lib/datastore/fragments/android/getting-started/configuration.md [hunk 2]
```diff
@@ -2,12 +2,27 @@
 
 In your app's `onCreate` (or similar lifecycle callback), initialize Amplify:
```
Comment (jpignata) [inline docs/lib/datastore/fragments/android/getting-started/configuration.md:3]: "Similar" is dubious advice to me - can we be prescriptive here?

File: docs/lib/datastore/fragments/android/getting-started/configuration.md [hunk 3]
```diff
@@ -2,12 +2,27 @@
...
 
+<amplify-block-switcher>
+<amplify-block name="Java">
+
 ```java
 Amplify.addPlugin(new AWSDataStorePlugin());
```
Comment (jpignata) [inline docs/lib/datastore/fragments/android/getting-started/configuration.md:9]: Let's do the whole try/catch dance here as people will copy/paste.

File: docs/lib/datastore/fragments/android/real-time/observe-snippet.md [hunk 4]
```diff
@@ -9,3 +12,21 @@ Amplify.DataStore.observe(Post.class,
     () -> Log.i("GetStarted", "Observation complete.")
 );
 ```
+
+</amplify-block>
+<amplify-block name="Kotlin">
+
+```kotlin
+Amplify.DataStore.observe(Post::class.java,
+    { Log.i("GetStarted", "Observation began.") },
```
Comment (jpignata) [inline docs/lib/datastore/fragments/android/real-time/observe-snippet.md:21]: Can we change these tags to `MyAmplifyApp` across this pull? I'd like to my them uniform across the whole Android section.

File: docs/lib/datastore/fragments/android/getting-started/configuration.md (hunk 2 above)
Comment (raphkim) [reply docs/lib/datastore/fragments/android/getting-started/configuration.md]: fixed

File: docs/lib/datastore/fragments/android/real-time/observe-snippet.md (hunk 4 above)
Comment (raphkim) [reply docs/lib/datastore/fragments/android/real-time/observe-snippet.md]: fixed

File: docs/lib/datastore/fragments/android/getting-started/configuration.md (hunk 3 above)
Comment (raphkim) [reply docs/lib/datastore/fragments/android/getting-started/configuration.md]: fixed

File: docs/lib/datastore/fragments/android/real-time/observe-snippet.md (hunk 1 above)
Comment (raphkim) [reply docs/lib/datastore/fragments/android/real-time/observe-snippet.md]: yessir it does

**Pull request merged**